File: data/multilang_comparison/program_cpp_vector/src/func49.cpp
#include "program_cpp_vector.hpp" 
Array* func49(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array15;
   if (pCounter > 0) {
      array15 = vars->data.at(--pCounter);
      array15->refC++;
      DEBUG_COPY(array15->id);
   } else {
      array15 = new Array();
      array15->size = 873;
      array15->refC = 1;
      array15->id = 15;
      array15->data = std::vector<unsigned int>(array15->size, 0);
      DEBUG_NEW(array15->id);
   }
   return array15;
}

File: data/multilang_comparison/program_go/program_go/src/func49.go
package main
import "math/rand"
var _ = rand.Intn
func Func49(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array15 *Array
   if pCounter > 0 {
       array15 = vars.Data[pCounter-1]
       pCounter--
       array15.RefC++
   } else {
       array15 = &Array{
           Size: 873,
           RefC: 1,
           Id: 15,
           Data: make([]uint32, 873),
       }
   }
   return array15;
}

